cloud/ec2.go

```diff
@@ -487,6 +487,64 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 	return h, nil
 }
 
+func (m *ec2Manager) ModifyHost(ctx context.Context, h *host.Host, changes host.HostModifyOptions) error {
+	ec2Settings := &EC2ProviderSettings{}
+	err := ec2Settings.fromDistroSettings(h.Distro)
+	if err != nil {
+		return errors.Wrap(err, "error getting EC2 settings")
+	}
+	if err := m.client.Create(m.credentials, ec2Settings.getRegion()); err != nil {
+		return errors.Wrap(err, "error creating client")
+	}
+	defer m.client.Close()
+
+	// note: deal with spot instances?
+	instanceID := h.Id
+	if isHostSpot(h) {
+		instanceID, err = m.client.GetSpotInstanceId(ctx, h)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get spot request info for %s", h.Id)
+		}
+		if instanceID == "" {
+			return errors.WithStack(errors.New("spot instance does not yet have an instanceId"))
+		}
+	}
+	volumeIDs, err := m.client.GetVolumeIDs(ctx, h)
+	if err != nil {
+		return errors.Wrapf(err, "can't get volume IDs for '%s'", h.Id)
+	}
+	resources := []string{instanceID}
+	resources = append(resources, volumeIDs...)
+
+	// Delete tags
+	deleteTagSlice := []*ec2.Tag{}
+	for _, key := range changes.DeleteInstanceTags {
+		deleteTagSlice = append(deleteTagSlice, &ec2.Tag{Key: &key})
```

Comment:
Should we verify that the tag exists in the host's instance tags before adding to delete?

Reply:
According to AWS documentation, it should be fine if a passed tag name does not exist.